src/actions/transactionEstimateActions.js

```diff
@@ -195,6 +209,14 @@ export const estimateTransactionsAction = (
       return;
     }
 
+    if (isKeyBasedAccount(activeAccount) && transactions.length > 1) {
+      reportErrorLog(
+        'estimateTransactionsAction failed: too many transactions for key based estimation',
+        { chain, transactions: transactions.length },
+      );
+      return;
+    }
```

Comment:
You're right that this won't happen in any other flow except Swaps, however, logic here is to better be safe than sorry since key based estimation can only accept single transaction while wrapped estimation action accepts multiple. Hope this makes sense to leave 👀

Reply:
Got it - thanks!